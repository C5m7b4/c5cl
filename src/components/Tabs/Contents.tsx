import React from 'react';

export interface ContentsProps {
  id: string;
  activeTab: string;
  children: React.ReactNode;
}

const Contents = (props: ContentsProps) => {
  const { id, activeTab, children } = props;

  return activeTab === id ? <div>{children}</div> : null;
};

export default Contents;
