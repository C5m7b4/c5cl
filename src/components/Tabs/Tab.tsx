import React from 'react';

export interface TabProps {
  className: string;
  onClick: (e: string) => void;
  title: string;
  activeTab: string;
  disabled: boolean;
  id: string;
}

const Tab = (props: TabProps) => {
  let className = 'tab-list-item';
  if (props.id === props.activeTab) {
    className += ' tab-list-active';
  }

  return (
    <li
      onClick={() => props.onClick(props.id)}
      className={className}
      id={props.id}
    >
      {props.title}
    </li>
  );
};

export default Tab;
