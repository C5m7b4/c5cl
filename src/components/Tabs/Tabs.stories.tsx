import React from 'react';
import { Story, Meta } from '@storybook/react';

import Tabs, { TabsProps } from './Tabs';

import { Tab1, Tab2 } from './Tab.stories';

export default {
  title: 'Tabs/Tabs',
  component: Tabs,
} as Meta;

const Template: Story<TabsProps> = (args) => (
  <Tabs>
    <div title="tab1">
      <p>Here is some content</p>
    </div>
    <div title="tab2">
      <p>Here is some more content</p>
    </div>
  </Tabs>
);

export const TabsExample = Template.bind({});
TabsExample.args = {
  // @ts-ignore
  children: [<Tab1 {...Tab1.args} />, <Tab2 {...Tab2.args} />],
};

// export const Tabs1Example = Template.bind({});
// Tabs1Example.args = {
//   children: [
//     // @ts-ignore
//     <div label="Tab1">
//       <h2>I am tab 1</h2>
//     </div>,
//     // @ts-ignore
//     <div label="Tab2">
//       <h2>I am tab 2</h2>
//     </div>,
//   ],
// };
