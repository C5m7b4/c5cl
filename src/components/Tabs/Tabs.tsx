import React, { useState, useEffect } from 'react';
import Tab from './Tab';
import Contents from './Contents';

import './Tabs.css';

export interface TabsProps {
  children: JSX.Element[];
}

const Tabs = (props: TabsProps) => {
  const [activeTab, setActiveTab] = useState<string>('');

  useEffect(() => {
    setActiveTab('tab-0');
  }, []);

  const handleTabClick = (tab: string) => {
    setActiveTab(tab);
  };

  return (
    <div className="tabs">
      <ol className="tab-list">
        {props.children &&
          props.children.map((child, index) => {
            if (child) {
              return (
                <Tab
                  activeTab={activeTab}
                  key={`tabs-li-${index}`}
                  title={child.props.title}
                  disabled={child.props.disabled}
                  onClick={handleTabClick}
                  id={`tab-${index}`}
                  className={child.props.className}
                />
              );
            }
          })}
      </ol>
      <div className="tab-content">
        {props.children &&
          props.children.map((child, idx) => (
            <Contents
              key={`tbc-${idx}`}
              id={`tab-${idx}`}
              activeTab={activeTab}
            >
              {child.props.children}
            </Contents>
          ))}
      </div>
    </div>
  );
};

export default Tabs;
