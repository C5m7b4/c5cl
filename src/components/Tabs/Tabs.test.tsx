// @ts-nocheck

import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Tabs from './Tabs';

describe('Tabs', () => {
  test('should render correctly', () => {
    const { container } = render(
      <Tabs>
        <div name="tab1" label="Tab1">
          <h2>Tab1</h2>
        </div>
        <div name="tab2" label="tab2">
          <h2>Tab2</h2>
        </div>
      </Tabs>
    );
    expect(container).toMatchSnapshot();
  });
  test('should handle click', () => {
    const { container, getByRole } = render(
      <Tabs>
        <div className="tab1" label="Tab1">
          <h2>Tab1</h2>
        </div>
        <div className="tab2" label="tab2">
          <h2>Tab2</h2>
        </div>
      </Tabs>
    );
    const tab = container.querySelector('div > div:nth-child(1)');
    // @ts-ignore
    fireEvent.click(tab);
  });
});
