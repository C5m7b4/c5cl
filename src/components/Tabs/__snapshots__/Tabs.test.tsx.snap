// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render correctly 1`] = `
<div>
  <div
    class="tabs"
  >
    <ol
      class="tab-list"
    >
      <li
        class="tab-list-item tab-list-active"
        id="tab-0"
      />
      <li
        class="tab-list-item"
        id="tab-1"
      />
    </ol>
    <div
      class="tab-content"
    >
      <div>
        <h2>
          Tab1
        </h2>
      </div>
    </div>
  </div>
</div>
`;
